Homepage Buttons

lewibt01 opened the pull request.

Commit:
wip adding control of optional client-only word wrap for codemirror

Commit:
wip: passed a higher level function to the toggle to facilitate word wrap

the props seem to be passing, but updating the actual codeCollab window is still a work in progress.

Commit:
Added 2 Buttons to the homepage to allow easy creating and rejoining collabs

Commit:
Merge branch 'test' into wordWrap

RobertMcCoy commented on package.json:
```diff
@@ -60,6 +60,7 @@
     "react-dom": "^16.0.0",
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
+    "react-scrollbar": "^0.5.1",
```
Remove unnecessary include

lewibt01 replied:
npm install wiped it out on my local repo. I don't see it in the package.json anymore, and the change should come through with my next commit.

RobertMcCoy replied:
Sounds good, it should wipe out the other references that came in then

RobertMcCoy commented on src/Collab.js:
```diff
@@ -151,6 +153,13 @@ class Collab extends Component {
+        this.setState({
+            options: {lineNumbers: true, mode: this.mode, lineWrapping: true}
+        });
+        this.addNotificationAlert("Line-Wrapping has been set to: "+ this.lineWrapping);
```
This shouldn't be an alert -- also where is this.lineWrapping coming from?

lewibt01 replied:
The alert was for debug, it will be gone in my next commit. This is in-progress code for word wrapping in codemirror. I opted to submit some of my changes so I could get feedback instead of spending 2-3 hours picking through my code to separate the wordWrap code from the homepage code. I'll create a separate branch for it once I fix the issues with the homepage.

RobertMcCoy commented on src/LandingPage.css:
```diff
@@ -28,16 +28,84 @@ h1, h3 {
+    border-radius: 8px;
+}
+
+.centered{
```
This can become margin: 0 auto

lewibt01 replied:
I've removed this, bootstrap css has made the .centered class obsolete

RobertMcCoy commented on src/LandingPage.css:
```diff
@@ -28,16 +28,84 @@ h1, h3 {
+    width:80%;
+}
+
+.fake-button-green{
```
Use bootstrap button styling

lewibt01 replied:
I didn't realize we could use bootstrap. That makes things way easier. I'll make the changes.

RobertMcCoy commented on src/LandingPage.js:
```diff
@@ -8,12 +10,24 @@ class LandingPage extends Component {
         this.state={
 
         }
+        var inputString = '';
```
This needs to be a property of the state object

lewibt01 replied:
fixed, input is now stored in the state object.

RobertMcCoy commented on src/LandingPage.js:
```diff
@@ -8,12 +10,24 @@ class LandingPage extends Component {
+                  <div className="fake-button-green centered" onClick={this.getName}>Start a Collab</div>
+                </Link>
+                <p className='bar'>|</p>
+                <input className='section-info round-input' ref={(input) => {this.inputString=input}} placeholder='Your Collab ID...'/>
```
Update this to pull from state instead of a local variable

lewibt01 replied:
fixed this at the same time that I changed the value to be stored in the state object.

RobertMcCoy commented on src/Toggle.css:
```diff
@@ -0,0 +1,27 @@
+.room-info {
```
A lot of classes are unused in this -- this looks like a copy of RoomInfo?

lewibt01 replied:
It doesn't look like toggle.js uses any of the classes in its css file, should I remove Toggle.css entirely?

RobertMcCoy replied:
Yea, use RoomInfo.css instead as that handles the `color: white` case

lewibt01 replied:
Removed Toggle.css and imported RoomInfo.css instead. Should appear in my next commit.

Commit:
Removed Unnecessary Toggle.css, updated landingPage.css to use bootstrap css, removed unnecessary import, wip: converting local variable to a state of landingPage.js

Commit:
Removed an unnecessary alert when attempting to handle lineWrapping...

re-added the .whiteText class to RoomInfo.css, 

Fixed "rejoin collab" button on landingpage.cs, 

Converted a local value to a state of landingpage.cs

wip: handle user-specified line wrapping

Commit:
wip: creating a callback function to pass a boolean from toggle.js to collab.js

RobertMcCoy committed:
merge: test to branch

RobertMcCoy committed:
fix: removed unecessary callback code, cleaned up update callback
